lib/neovim-plugin.nix

```diff
@@ -152,13 +159,62 @@
                     }
                   ))
                 ]
-                ++ (lib.optionals (!hasConfigAttrs) [
+                ++ lib.optionals (!hasConfigAttrs) [
                   (lib.optionalAttrs callSetup (setLuaConfig setupCode))
-                ])
-                ++ (lib.optionals hasConfigAttrs [
+                ]
+                ++ lib.optionals hasConfigAttrs [
                   (lib.optionalAttrs callSetup { ${namespace}.${name}.luaConfig.content = setupCode; })
-                  (lib.optionalAttrs (configLocation != null) (setLuaConfig cfg.luaConfig.content))
-                ])
+                  (lib.mkIf (!cfg.lazyLoad.enable) (
+                    lib.optionalAttrs (configLocation != null) (setLuaConfig cfg.luaConfig.content)
+                  ))
+                ]
+                ++ lib.optionals hasConfigAttrs [
+                  (lib.mkIf cfg.lazyLoad.enable {
+                    assertions = [
+                      {
+                        assertion = (isColorscheme && colorscheme != null) || cfg.lazyLoad.settings != { };
+                        message = "You have enabled lazy loading for ${originalName} but have not provided any configuration.";
+                      }
+                      {
+                        assertion =
+                          lib.length (
+                            lib.filter (x: config.plugins.${x}.enable) [
+                              "lz-n"
+                              "lazy"
+                            ]
+                          ) == 1;
+                        message = "You have enabled multiple lazy providers. Enable only one.";
+                      }
```

Comment:
My mistake: this should not be a per-plugin assertion.

Instead, we should probably have a global "lazyload" module somewhere that defines this assertion only once.

Perhaps such a module could end up being the future home of any _global_ lazyloading options too?

Might tackle that in a separate PR, since it shouldn't depend on anything in this one.

Reply:
https://github.com/nix-community/nixvim/pull/2609